File: Modules/ServerTypes/BringForward/BringForwardForm.ts
import { IntegerEditor, DateEditor, PrefixedContext } from "@serenity-is/corelib";
import { initFormType } from "@serenity-is/corelib/q";

export interface BringForwardForm {
    DeleteUserId: IntegerEditor;
    UpdateUserId: IntegerEditor;
    InsertUserId: IntegerEditor;
    IsActive: IntegerEditor;
    InsertDate: DateEditor;
    UpdateDate: DateEditor;
    DeleteDate: DateEditor;
    EmployeeRowId: IntegerEditor;
    BringForward: IntegerEditor;
    BringForwardToYear: IntegerEditor;
}

export class BringForwardForm extends PrefixedContext {
    static formKey = 'BringForward.BringForward';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!BringForwardForm.init)  {
            BringForwardForm.init = true;

            var w0 = IntegerEditor;
            var w1 = DateEditor;

            initFormType(BringForwardForm, [
                'DeleteUserId', w0,
                'UpdateUserId', w0,
                'InsertUserId', w0,
                'IsActive', w0,
                'InsertDate', w1,
                'UpdateDate', w1,
                'DeleteDate', w1,
                'EmployeeRowId', w0,
                'BringForward', w0,
                'BringForwardToYear', w0
            ]);
        }
    }
}

File: Modules/HumanResource/BringForward/BringForward/BringForwardForm.cs
using Serenity.ComponentModel;
using System;

namespace HRMSoftware.BringForward.Forms;

[FormScript("BringForward.BringForward")]
[BasedOnRow(typeof(BringForwardRow), CheckNames = true)]
public class BringForwardForm
{
    public int DeleteUserId { get; set; }
    public int UpdateUserId { get; set; }
    public int InsertUserId { get; set; }
    public int IsActive { get; set; }
    public DateTime InsertDate { get; set; }
    public DateTime UpdateDate { get; set; }
    public DateTime DeleteDate { get; set; }
    public int EmployeeRowId { get; set; }
    public int BringForward { get; set; }
    public int BringForwardToYear { get; set; }
}